src/commcare_cloud/ansible/roles/airflow/tasks/main.yml

```diff
@@ -0,0 +1,60 @@
+- name: create required directories
+  become: yes 
+  file:
+    path: "{{ item }}"
+    owner: "{{ cchq_user }}"
+    group: "{{ cchq_user }}"
+    mode: 0755
+    state: directory
+  with_items:
+    - "{{ airflow_home }}"
+    - "{{ airflow_code_dir }}"
+
+- name: Pull full Airflow source
+  git:
+    repo: "{{ airflow_repository }}"
+    dest: "{{ airflow_code_dir }}"
+    version: "{{ airflow_version }}"
+    recursive: yes
+    accept_hostkey: yes
+    update: yes
+  become: yes
+  become_user: "{{ cchq_user }}"
+
+- name: install pip requirements
+  become: yes
+  become_user: "{{ cchq_user }}"
+  pip:
+    requirements: "{{ airflow_code_dir }}/requirements.txt"
+    virtualenv: "{{ airflow_virtualenv }}"
+    chdir: "{{ airflow_code_dir }}"
+
+- name: copy airflow settings
```

Comment:
Ran into this during a search. I think (at least eventually) this task should get run on `update-config`, which also updates `localsettings.py` and formplayer's `application.properties`.

Reply:
I agree with some of that but I also think that it's useful to have a command to update all settings - for instance if you change the IP of a service that they all use.

Perhaps the best approach would be to make the `update-config` command able to update the all or each one individually.

Reply:
Yeah, I am not opposed to including this in `update-config` as long as a dedicated way still exists. In the past we have replaced those methods with just adding it to `update-config`, so there is no longer an officially supported way to deploy just localsettings for example, which is probably the dominant case.  In the same way we separate `service restart` to happen by service, even though sometimes you want to do all of them, I think it makes sense to maintain the ability to update single configs.

I more meant to call out what I saw as a general trend of grouping of tasks that are conceptually similar but unlikely to be needed at the same time into single commands, as a replacement for the individual ones.